src/watertap_contrib/reflo/analysis/multiperiod/vagmd_batch/VAGMD_batch_multiperiod_flowsheet.py

```diff
@@ -0,0 +1,495 @@
+    m.fs.acc_distillate_volume.unfix()
+    m.fs.acc_thermal_energy.unfix()
+    m.fs.pre_thermal_power.unfix()
+    m.fs.acc_cooling_energy.unfix()
+    m.fs.pre_cooling_power.unfix()
+    m.fs.acc_electric_energy.unfix()
+    m.fs.pre_feed_pump_power_elec.unfix()
+    m.fs.pre_cooling_pump_power_elec.unfix()
+
+    m.fs.vagmd.feed_props[0].temperature.unfix()
+    m.fs.vagmd.feed_props[0].flow_mass_phase_comp["Liq", "TDS"].unfix()
+    m.fs.vagmd.feed_props[0].flow_mass_phase_comp["Liq", "H2O"].unfix()
+    m.fs.vagmd.feed_props[0].flow_vol_phase["Liq"].fix(
+        pyunits.convert(
+            feed_flow_rate * pyunits.L / pyunits.h, to_units=pyunits.m**3 / pyunits.s
+        )
+    )
+
+    return
+
+
+def build_VAGMD_batch_multiperiod_fs(
+    m,
+    n_time_points,
+    system_capacity=1000,  # m3/day
+    # Arguments below are the initialized condition for multiperiod model construction,
+    dt=None,
+    feed_flow_rate=600,  # 400 - 1100 L/h
+    evap_inlet_temp=80,  # 60 - 80 deg C
+    cond_inlet_temp=25,  # 20 - 30 deg C
+    feed_temp=25,  # 20 - 30 deg C
+    feed_salinity=35,  # 35 - 292 g/L
+    # initial_batch_volume=50,  # > 50 L
+    # recovery_ratio=0.5,  # -
+    module_type="AS7C1.5L",
+    cooling_system_type="closed",
+    cooling_inlet_temp=25,  # 20 - feed_temp deg C, not required when cooling system type is "closed"
+):
+    """
+    This function builds a multiperiod flowsheet of the VAGMD system
+    for the simulation of a batch operation
+
+    Returns:
+        object: A multiperiod module created for the target Pyomo optimization model m
+    """
+
+    if m is None:
+        m = ConcreteModel()
+
+    # Create the multiperiod model object
+    m.mp = MultiPeriodModel(
+        n_time_points=n_time_points,
+        process_model_func=build_vagmd_flowsheet,
+        linking_variable_func=get_vagmd_batch_variable_pairs,
+        initialization_func=fix_dof_and_initialize,
+        unfix_dof_func=unfix_dof,
+        outlvl=logging.WARNING,
+    )
+
+    mp = m.mp
+
+    """
+    Specify the initialization conditions of each period
+    """
+
+    # Model options listed in the VAGMD flowsheet of a single period,
+    # see arguments from the imported 'build_vagmd_flowsheet' method.
+    model_options = {
+        "dt": dt,
+        "system_capacity": system_capacity,
+        "feed_flow_rate": feed_flow_rate,
+        "evap_inlet_temp": evap_inlet_temp,
+        "cond_inlet_temp": cond_inlet_temp,
+        "feed_temp": feed_temp,
+        "feed_salinity": feed_salinity,
+        "module_type": module_type,
+        "cooling_system_type": cooling_system_type,
+        "cooling_inlet_temp": cooling_inlet_temp,
+    }
+
+    mp.build_multi_period_model(
+        model_data_kwargs={t: model_options for t in range(n_time_points)},
+        flowsheet_options=model_options,
+        initialization_options={
+            "feed_flow_rate": feed_flow_rate,
+            "feed_salinity": feed_salinity,
```

Comment:
Feed salinity and flow rates are not fixed in the fix_dof_and_initialize function currently. Do we still need this here in that case?

Reply:
I'd say this is a question for @Zhuoran29

Reply:
though @MuktaHardikar I see that tests were passing before and after your change in f383125 so wondering why you wanted to release state?

Reply:
@MuktaHardikar  No, those two redundant arguments were never used in the initialization routine. Now I removed them in all associated functions. Thanks!